src/main/java/com/broadleafcommerce/autoconfigure/HSQLDBServer.java

```diff
@@ -64,20 +68,62 @@ public HSQLDBServer(final HSQLDBProperties autoProps, Environment environment) {
 
     @Override
     public boolean isRunning() {
-        try (Socket ignored = new Socket(InetAddress.getByName(null), props.getIntegerProperty("server.port", 0))) {
-            return true;
-        } catch (IOException ignored) {
-            return false;
+        boolean isRunning = false;
+        Connection c = null;
+
+        final int port = props.getIntegerProperty("server.port", 0);
+        try {
+            Class.forName("org.hsqldb.jdbc.JDBCDriver");
+            final String url = "jdbc:hsqldb:hsql://" + InetAddress.getByName(null).getHostName() + ":"
+                               + port + "/"
+                               + props.getProperty("server.dbname.0", "");
+            final String username = "SA";
+            final String password = "";
+            
+            c = DriverManager.getConnection(url, username, password);
+            
+            isRunning = true;
+        } catch (ClassNotFoundException e) {
+            LOG.error("ERROR: Cannot test if an HSQL server is running: failed to load HSQLDB JDBC driver.", e);
+        } catch (SQLException e) {
+            Socket s = null;
+            try {
+                // see if the port is being used already (by something other than HSQL)
+                s = new Socket(InetAddress.getByName(null), port);
+                LOG.error("Port," + port + ", is already in use but not by HSQL");
+            } catch (Exception ignored) {
+                // otherwise, it's not in use, yet
+                LOG.info("HSQL server is not running.");
+            } finally {
+                closeConnectionQuietly(s);
+            }
+        } catch (UnknownHostException e) {
+            LOG.error("ERROR: An error occurred while testing whether an HSQL server is running.", e);
+        } finally {
+            closeConnectionQuietly(c);
+        }
+        
+        return isRunning;
+    }
+    
+    protected void closeConnectionQuietly(AutoCloseable c) {
```

Comment:
Previously, I was using the Java 7-compatible [try with resource](https://www.mkyong.com/java/try-with-resources-example-in-jdk-7/) that does exactly what you're doing here with anything that implements `AutoCloseable`. Do that instead of this extra call in a `finally` block.